servers/src/main/java/tachyon/worker/block/allocator/RoundRobinAllocator.java

```diff
@@ -0,0 +1,120 @@
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.allocator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.BlockMetadataManagerView;
+import tachyon.worker.block.meta.StorageDirView;
+import tachyon.worker.block.meta.StorageTierView;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * A Round robin allocator that allocates a block in the storage dir.
+ * It will allocate the block in the highest tier possible:
+ * It always starts from the highest tier in a RR manner and goes to the next tier when there
+ * is no enough space. 
+ * The allocator only considers non-specific writes in its RR policy (the location is either 
+ * AnyTier or AnyDirInTier).
+ */
+public class RoundRobinAllocator implements Allocator {
+  private BlockMetadataManagerView mManagerView;
+
+  // We need to remember the last dir index for every storage tier
+  private Map<StorageTierView, Integer> mTierViewDirViews = new HashMap<StorageTierView, Integer>();
+
+  public RoundRobinAllocator(BlockMetadataManagerView view) {
+    mManagerView = Preconditions.checkNotNull(view);
+    
+    for (StorageTierView tierView : mManagerView.getTierViews()) {
+      mTierViewDirViews.put(tierView, -1);
+    }
+  }
+
+  @Override
+  public TempBlockMeta allocateBlockWithView(long userId, long blockId, long blockSize,
+      BlockStoreLocation location, BlockMetadataManagerView view) throws IOException {
+    mManagerView = view;
+    return allocateBlock(userId, blockId, blockSize, location);
+  }
+
+  /**
+   * Should only be accessed by {@link allocateBlockWithView} inside class.
+   * Allocates a block from the given block store location. The location can be a specific location,
+   * or {@link BlockStoreLocation#anyTier()} or {@link BlockStoreLocation#anyDirInTier(int)}.
+   *
+   * @param userId the ID of user to apply for the block allocation
+   * @param blockId the ID of the block
+   * @param blockSize the size of block in bytes
+   * @param location the location in block store
+   * @return a temp block meta if success, null otherwise
+   * @throws IOException if block location is invalid
+   */
+  private TempBlockMeta allocateBlock(long userId, long blockId, long blockSize,
+      BlockStoreLocation location) throws IOException {
+    if (location.equals(BlockStoreLocation.anyTier())) {
+      int tierAlias = 0; //always starting from the first tier
```

Comment:
The way you use `tierAlias` here (you called `.get(tierAlias)` at line 76) is actually `tierIndex`. I would try avoid using `tierAlias` as it is confusing..